### ISSUE TITLE
"'unicodeescape' codec can't decode bytes in position 3-9: illegal Unicode character" when using guess_lexer

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#912]
(Original issue 1205 created by tmortagne on 2016-01-29T17:07:37.888393+00:00)

I upgraded from 2.0.2 to 2.1 and my code that use guess_lexer ends up with:

```
#!

Caused by: Traceback (most recent call last):
  File "<script>", line 35, in <module>
  File "<script>", line 35, in <module>
  File "__pyclasspath__/pygments/lexers/__init__.py", line 245, in guess_lexer
  File "__pyclasspath__/pygments/lexers/__init__.py", line 174, in _iter_lexerclasses
  File "__pyclasspath__/pygments/lexers/__init__.py", line 41, in _load_lexers
UnicodeDecodeError: 'unicodeescape' codec can't decode bytes in position 3-9: illegal Unicode character
```

My code is pretty simple:

```
#!python

pygmentLexer = guess_lexer('toto', stripnl=False)
```

Usually it's an input but I tried with 'toto' to be really sure it did not came from my input and I get the exact same error.

I'm using Pygments trough Jython 2.7 so it might be a Jython issue too but guess_lexer used to work well with the same version of Jython so at least you might have an idea of what it does not like.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2016-02-02T09:32:58.958249+00:00)

The traceback is not very helpful unfortunately, but the source of this is usually that a Unicode string contains raw surrogate characters, which Jython does not allow in Unicode strings. I found the change that caused this and reverted it now.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2016-02-02T09:33:22.930579+00:00)

Backout 10520a0c4913 - this breaks Jython.

If you want to handle non-BMP characters correctly with Python 2.x,
you should use a build with wide unicode characters.

Closes #1205.

→ <<cset 9dd506d3bc99>>
